src/Appwrite/Platform/Workers/StatsUsage.php

```diff
@@ -246,8 +323,126 @@ private function reduce(Document $project, Document $document, array &$metrics,
+
+                    foreach ($this->periods as $period => $format) {
+                        $time = null;
+
+                        if ($period !== 'inf') {
+                            $time = !empty($receivedAt) ? (new \DateTime($receivedAt))->format($format) : date($format, time());
+                        }
+                        $id = \md5("{$time}_{$period}_{$key}");
+
+                        $document = new Document([
+                            '$id' => $id,
+                            'period' => $period,
+                            'time' => $time,
+                            'metric' => $key,
+                            'value' => $value,
+                            'region' => System::getEnv('_APP_REGION', 'default'),
+                        ]);
+
+
+                        $this->projects[$project->getInternalId()]['project']  = new Document([
+                            '$id' => $project->getId(),
+                            '$internalId' => $project->getInternalId(),
+                            'database' => $project->getAttribute('database'),
+                        ]);
+                        $this->projects[$project->getInternalId()]['stats'][] = $document;
+
+                        $this->prepareForLogsDB($project, $document);
+                    }
+                }
+            } catch (Exception $e) {
+                console::error('[' . DateTime::now() . '] project [' . $project->getInternalId() . '] database [' . $project['database'] . '] ' . ' ' . $e->getMessage());
+            }
+        }
+
+        foreach ($this->projects as $internalId => $projectStats) {
+            if (empty($internalId)) {
+                continue;
+            }
+            try {
+                /** @var \Utopia\Database\Database $dbForProject */
+                $dbForProject = $getProjectDB($projectStats['project']);
+                Console::log('Processing batch with ' . count($projectStats['stats']) . ' stats');
+                $dbForProject->createOrUpdateDocumentsWithIncrease('stats', 'value', $projectStats['stats']);
+                Console::success('Batch successfully written to DB');
+
+                unset($this->projects[$internalId]);
+            } catch (Throwable $e) {
+                Console::error('Error processing stats: ' . $e->getMessage());
+            }
+        }
+
+        $this->writeToLogsDB();
+
+    }
+
+    protected function prepareForLogsDB(Document $project, Document $stat)
+    {
+        if (System::getEnv('_APP_STATS_USAGE_DUAL_WRITING', 'disabled') === 'disabled') {
+            return;
+        }
+        if (array_key_exists($stat->getAttribute('metric'), $this->skipBaseMetrics)) {
+            return;
+        }
+        foreach ($this->skipParentIdMetrics as $skipMetric) {
+            if (str_ends_with($stat->getAttribute('metric'), $skipMetric)) {
+                return;
+            }
+        }
+        $documentClone = new Document($stat->getArrayCopy());
```

Comment:
We have __clone defined on document so you could do `$documentClone = clone $stat;` here

Reply:
updated